### ISSUE TITLE
training loss at each iteration is zero when take train_numerical_elwc.tfrecord as input for tf_ranking_tfrecord.py

### ISSUE THREAD

**Issue by twinsken** [tensorflow/ranking#213]
Hi, I am trying to use the dense feature as input to train an unbiased ranker. 

Since the provided example tf_ranking_tfrecord.py only support sparse feature like query_token, doc_token. To achieve this goal, I use the input train_numerical_elwc.tfrecord as an example and feed it to tf_ranking_tfrecord.py. Besides, I modify the function **example_feature_columns** and **context_feature_columns** based on the implementation of **tf_ranking_canned_dnn.py**.
`
def context_feature_columns():
  """Returns context feature columns."""
  return {}

def example_feature_columns(use_weight_feature=True):
  feature_names = [
      "custom_features_{}".format(i + 1) for i in range(FLAGS.num_features)
  ]
  return {
      name:
      tf.feature_column.numeric_column(name, shape=(1,), default_value=0.0)
      for name in feature_names
  }`

However, the results show that the training loss at each iteration is 0. The final NDCG on eval data is 0.05.

Another question:  is there any way to make tf_ranking_libsvm.py support weights_feature_name which is used for unbiased learning-to-rank?

**Comment by HongleiZhuang** [comment]
Hi @twinsken 

I tried to do similar things but the training seems to work with NDCG@5 around 0.758.  Can you please check whether you did the following steps:

1. Add the `num_features` flag by adding this line:
`flags.DEFINE_integer("num_features", 136, "Number of features per example.")`

2. Change the `_LABEL_FEATURE` to "utility":
`_LABEL_FEATURE = "utility"`

Regarding your second question, you may take a look at how `weights_feature_name` is used in [loss function](https://github.com/tensorflow/ranking/blob/master/tensorflow_ranking/examples/tf_ranking_tfrecord.py#L376)
and [metrics](https://github.com/tensorflow/ranking/blob/master/tensorflow_ranking/examples/tf_ranking_tfrecord.py#L352) in **tf_ranking_tfrecord.py**, and change **tf_ranking_libsvm.py** accordingly.

**Comment by xuanhuiwang** [comment]
close this due to inactivity.
